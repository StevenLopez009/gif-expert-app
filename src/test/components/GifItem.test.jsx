import GifItem from "../../components/GifItem";
import { render, screen } from "@testing-library/react";

describe("Pruebas en <GifItem/>", () => {
  const title = "Jinx";
  const url = "http://arcane.com/";
  test("hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("debe de mostrar la imagen con el url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);
    //expect(screen.getByRole("img").src).toBe(url);
    //expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });
});
