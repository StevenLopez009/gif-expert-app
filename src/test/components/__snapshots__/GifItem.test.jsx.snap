// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Jinx"
      src="http://arcane.com/"
    />
    <p>
      Jinx
    </p>
  </div>
</div>
`;
